### DISCUSSION TITLE
how to using Lit v2 elements in Polymer 3 app?

### DISCUSSION THREAD

**Discussion by prakash-npm** [lit/lit#2928] [Q&A] [answered]
I have converted a Polymer 2 app to Polymer 3 instead of LIT v2, because of the size of the project. I am using Lit v1 elements inside Polymer 3 without any problem. Even Lit v2 elements are working fine when I define Reactive properties as

// similar to Lit v1

static get properties() { return { propertyName: options }; }

when I change those to Lit v2

static properties = {
  propertyName: {type: String}
};

'polymer serve -o' throws the following error: Uncaught TypeError: Failed to resolve module specifier "lit". Relative references must start with either "/", "./", or "../".

I have tried to change the Polymer 3 app shell component into Lit v2, and used 'polymer serve' above-mentioned reference error occurs. When I tried to use 'wds --watch' it throws the following error: Uncaught TypeError: Failed to resolve module specifier "@polymer/app-layout/app-drawer/app-drawer.js". Relative references must start with either "/", "./", or "../". I have added all the dependencies properly.

I am writing new components in Lit v1 only instead of Lit v2 because of this issue.

Can anyone please tell me where I'm going wrong in my thinking?

**Comment by kevinpschaaf** [comment]
Could you possibly make a minimal repo to share? It sounds like node module resolution in either `polymer serve` or `wds` is failing to find the package in your node_modules.

**Reply by prakash-npm** [reply]
Hi Kevin, 
Thanks for your reply.
I have created two repositories one with Polymer 3 shell project, and one I have upgraded to Lit v1. Please use these links:
https://github.com/prakash-npm/Using-Lit-v2-element-in-Polymer3
https://github.com/prakash-npm/Using-Lit-v2-element-in-Lit-v1

I have created 'my-sample.js' Lit v2 element, I have commented out the Lit v1 way of declaring both 'Style & Properties'. It works with Lit v1 way of declaration using 'polymer serve'.

-Prakash

**Reply by prakash-npm** [reply]
Hi Kevin,
When I tried to build the app 'https://github.com/prakash-npm/Using-Lit-v2-element-in-Polymer3' below error occurs:

../src/my-app.js(169,9) error [could-not-load] - Unable to load import: This experimental syntax requires enabling the parser plugin: 'classProperties' (18:17)
error:  Uncaught exception: Error: 1 error(s) occurred during build.
error:  Error: 1 error(s) occurred during build.
    at BuildAnalyzer._done (C:\Users\prakash.s\AppData\Roaming\npm\node_modules\polymer-cli\node_modules\polymer-build\lib\analyzer.js:277:36)
    at BuildAnalyzer.<anonymous> (C:\Users\prakash.s\AppData\Roaming\npm\node_modules\polymer-cli\node_modules\polymer-build\lib\analyzer.js:238:26)
    at Generator.next (<anonymous>)
    at fulfilled (C:\Users\prakash.s\AppData\Roaming\npm\node_modules\polymer-cli\node_modules\polymer-build\lib\analyzer.js:17:58)

I even tried to added a file '.babelrc' with the following setting, the same error I have mentioned above occurs.

 {
      "plugins": [
        "@babel/plugin-proposal-class-properties",
        "@babel/plugin-proposal-private-methods"
      ]
  }

**Comment by kevinpschaaf** [comment, marked as answer]
Yeah, so unfortunately the language features supported by `polymer-build` are [hard-coded](https://github.com/Polymer/tools/blob/2f5749613d1ecf5a396345b2888fe620b539e9e6/packages/analyzer/src/javascript/javascript-parser.ts#L34-L42), and the polymer CLI is no longer receiving feature updates, so it just won't support newer language features like field initializers (`static properties = {...}` vs. `static get properties() { return {...} }`.

If you want to stay with `polymer build`, you can just use `static get properties()`, etc.; practically speaking there's no real advantage to using fields vs. accessors other than it's a little cleaner. To the library, it makes no difference.

Alternatively, you could look into upgrading your build setup to something more modern. On [lit.dev, we have instructions](https://lit.dev/docs/tools/production/#modern-+-legacy-build) for building with rollup, including a build for both modern and older browsers that's comparable to what polymer build did.
